test/e2e/scenario/recorder/viewports.scenario.ts

```diff
@@ -0,0 +1,371 @@
+import { IncrementalSource, ViewportResizeData, ScrollData } from '@datadog/browser-rum/cjs/types'
+import { RumInitConfiguration } from '@datadog/browser-rum-core'
+
+import { findAllIncrementalSnapshots, findAllVisualViewports } from '@datadog/browser-rum/test/utils'
+import { createTest, bundleSetup, html, EventRegistry } from '../../lib/framework'
+import { browserExecute } from '../../lib/helpers/browser'
+import { flushEvents } from '../../lib/helpers/sdk'
+
+const VIEWPORT_META_TAGS = `
+<meta name="apple-mobile-web-app-capable" content="yes">
+<meta name="mobile-web-app-capable" content="yes">
+<meta name="viewport"
+  content="width=device-width, initial-scale=1.0, maximum-scale=2.75, minimum-scale=1.0, user-scalable=yes"
+>
+`
+
+describe('recorder', () => {
+  beforeEach(() => {
+    if (isGestureUnsupported()) {
+      pending('no touch gesture support')
+    }
+  })
+
+  describe('layout viewport properties', () => {
+    createTest('getWindowWidth/Height should not be affected by pinch zoom')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        await buildScrollablePage()
+
+        const { innerWidth, innerHeight } = await getWindowInnerDimensions()
+        await performSignificantZoom()
+
+        await browserExecute(() => {
+          window.dispatchEvent(new Event('resize'))
+        })
+
+        const lastViewportResizeData = (
+          await getLastRecord(events, (segment) =>
+            findAllIncrementalSnapshots(segment, IncrementalSource.ViewportResize)
+          )
+        ).data as ViewportResizeData
+
+        const scrollbarWidth = await getScrollbarCorrection()
+
+        expectToBeNearby(lastViewportResizeData.width, innerWidth - scrollbarWidth)
+        expectToBeNearby(lastViewportResizeData.height, innerHeight - scrollbarWidth)
```

Comment:
```suggestion
        expectToBeNearby(lastViewportResizeData.height, innerHeight - scrollbarCorrection)
```
a bit weird to subtract a width from a height, no?

Reply:
Yeah. Correction is probably better than something like `scrollbarThickness`.